src/render/Renderer.cpp

```diff
@@ -1265,10 +1265,6 @@ void CHyprRenderer::renderMonitor(CMonitor* pMonitor) {
         pMonitor->forceFullFrames                      = 10;
     }
 
-    bool lockSoftware = pMonitor == g_pCompositor->getMonitorFromCursor() && *PZOOMFACTOR != 1.f;
```

Comment:
why not do it here?

```cpp
static bool zoomLock = false;
if (zoomlock && !zoom) {
    unlock();
    zoomlock = false;
} else if (!zoomlock && zoom) {
    lock();
    zoomlock = true;
}
```

Reply:
you right lol, i was thinking too hard

Reply:
wait actually. the reason is because we need to lock software cursors for all monitors or else hardware cursors breaks stuff by not disabling correctly